### DISCUSSION TITLE
[BE101] Wee12 HW1 載入更多按鈕功能

### DISCUSSION THREAD

**Discussion by ddylanlinn** [Lidemy/forum#151] [程式導師實驗計畫] [answered]
大家好
在這周跟著老師實做[MTR05] Week12 HW1 後有個小疑惑，

#### 功能：初始頁顯示5則留言，點擊`載入更多`按鈕則再顯示5則，直到沒有留言

老師影片範例的方法是，在後端php裡面設定一次請求只給取5筆資料，
若要再拿取資料，則依據 `GET` 裡 `before` 值去判斷，拿取幾號 id 之後的5筆
```php
SELECT * FROM comments WHERE site_key = XXX
  (empty($_GET['before']) ? '' : 'AND id < ?') ORDER BY id DESC LIMIT 5 '
```
前端則是每點擊一次發出一次請求（夾帶before值）

---
而我最開始找資料實做功能時，另一個方法是
只請求一次，載入頁面同時就載入所有資料

預設所有的留言為 display:none

執行JS下判斷
```js
$(".card").slice(0, 5).show()  // 先顯示前五筆資料
 $("#loadmore").click( e =>{  // 按鈕點擊監聽
  $(".card:hidden").slice(0, 5).show()  // 依序顯示隱藏的5筆 
  if($(".card:hidden").length == 0){  // 當無隱藏資料時隱藏按鈕  
   e.target.style.display = 'none'  
   }
 })
```

---
想請問這兩種方法的差別除了效能以外（一個只請求需要且足夠的資料量、另一個一次全部拿再做處理）
還有什麼其他的優劣差異嗎？
（第二種方法真的超快ㄉ🤣，是不是在確定資料量小的前提下可以這樣用？）

謝謝

**Comment by aszx87410** [comment, marked as answer]
第二個就比較不 scale，資料變多的時候還是要回到第一種，但優點就是只要一次 request
如果真的確定資料量很小的話，第二種也 ok
總之就是看資料量啦，大的話一定要第一種，小的話兩種都行
